addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as denodeify from 'denodeify';
+
+/**
+ * Declare an Interface where key is a string and value is either a string or a number.
+ * Use for storing a string literal, start position and end posittion of ImportClause node kinds.
+ */
+export interface ModuleImport {
+  specifier: string;
+  pos: number;
+  end: number;
+};
+
+export interface ModuleMap {
+  [key: string]: ModuleImport[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createTsSourceFile(fileName: string): Promise<ts.SourceFile> {
+  const readFile = denodeify(fs.readFile);
+  return readFile(fileName, 'utf8')
+    .then((contents: string) => {
+      return ts.createSourceFile(fileName, contents, ts.ScriptTarget.ES6, true);
+    });
+}
+
+/**
+ * @return traverses through ts.Node and returns an array of moduleSpecifiers
+ * 
+ * @param {ts.SourceFile} node: Typescript Node of whose AST is being traversed
+ * 
+ */
+export function getImportClauses(node: ts.SourceFile): ModuleImport[] {
+  let moduleSpecifiers: ModuleImport[] = [];
+  if (node.statements.length > 0) {
+    node.statements.forEach((nodeObj: ts.Node) => {
+      if (nodeObj.kind === ts.SyntaxKind.ImportDeclaration) {
+        nodeObj.getChildren().forEach((childNode: ts.Node) => {
+          if (childNode.kind === ts.SyntaxKind.StringLiteral) {
+            let specifierText = childNode.getFullText();
```

Comment:
Btw, `node.moduleSpecifier.text` provides the desired specifierText. I was having problem only when using `node.getFullText()` to get the text.